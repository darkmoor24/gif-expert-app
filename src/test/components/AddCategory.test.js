import React from "react";
// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AddCategory  from "../../components/AddCategory";
import { shallow } from "enzyme";

describe('Probando <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();

        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Probando cambios en el <input />', () => {
        const input = wrapper.find('input');
        const value = "Hola mundo";

        input.simulate("change", { target: { value } });
    });

    test('Probando onSubmit del formulario - No debe postear la informaciÃ³n', () => {
        wrapper.find("form").simulate("submit", { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = "Hola mundo";

        input.simulate("change", { target: { value } });

        wrapper.find("form").simulate("submit", { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );

        expect(input.text().trim()).toBe("");
    });
});