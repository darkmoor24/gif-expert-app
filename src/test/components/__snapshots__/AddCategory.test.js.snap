// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <AddCategory /> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
